libkbfs/folder_update_prepper.go

```diff
@@ -0,0 +1,1223 @@
+// Copyright 2017 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package libkbfs
+
+import (
+	"fmt"
+
+	"github.com/keybase/client/go/logger"
+	"github.com/keybase/client/go/protocol/keybase1"
+	"github.com/keybase/kbfs/kbfsblock"
+	"github.com/keybase/kbfs/tlf"
+	"github.com/pkg/errors"
+	"golang.org/x/net/context"
+)
+
+// folderUpdatePrepper is a helper struct for preparing blocks and MD
+// updates before they get synced to the backend servers.  It can be
+// used for a single update or for a batch of updates (e.g. conflict
+// resolution).
+type folderUpdatePrepper struct {
+	config       Config
```

Comment:
1. It makes explicit the dependencies this component takes.
1. It makes components more testable: you only have to implement test versions of the methods actually used (or pass in stock test implementations of each interface).

If the names bother you, you can explicitly name the methods:
```go
type folderUpdatePrepperConfig interface {
    Clock() Clock
    Codec() kbfscodec.Codec
    BlockCache() BlockCache
    ...etc
}
```
The `cryptoPure` change below should happen anyway though. All the methods that use `.Crypto()` take a `cryptoPure`, so there's no need to pass the full `Crypto`.